app/presenters/hyrax/work_show_presenter.rb

```diff
@@ -98,6 +98,9 @@ def representative_presenter
           result = member_presenters([representative_id]).first
           return nil if result.try(:id) == id
           result.try(:representative_presenter) || result
+        rescue ArgumentError # Raised by valkyrie if representative_id object is not indexed
```

Comment:
this seems like a heavy handed rescue. won't this catch and swallow any ArgumentError regardless of the cause?

i think more precise handling is needed.

Reply:
Yes, but that's the type of error raised by [PcdmMemberPresenterFactory#member_presenters](https://github.com/samvera/hyrax/blob/main/app/presenters/hyrax/pcdm_member_presenter_factory.rb#L64). We could change that to be a more unique error class.

Reply:
could we rework `SolrDocumentBehavior` so L95 `return nil if representative_id.blank?` handles this case?

Reply:
The case this change is trying to address is when representative_id contains the id of a deleted or not yet indexed object, so SolrDocument is going to have a representative_id it assumes exists. We are trying to cover both ends of the issue by cleaning up when removing a fileset and making the presenter side less fragile.

Reply:
i like the changes that have come through on this. thanks!